src/main/scala/monix/nio/tcp/AsyncTcpClientObservable.scala

```diff
@@ -0,0 +1,135 @@
+package monix.nio.tcp
+
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+
+import monix.eval.{Callback, Task}
+import monix.execution.Ack.{Continue, Stop}
+import monix.execution.atomic.Atomic
+import monix.execution.{Cancelable, UncaughtExceptionReporter}
+import monix.execution.cancelables.SingleAssignmentCancelable
+import monix.execution.exceptions.APIContractViolationException
+import monix.nio.{Bytes, EmptyBytes, NonEmptyBytes}
+import monix.nio.cancelables.SingleFunctionCallCancelable
+import monix.reactive.Observable
+import monix.reactive.observers.Subscriber
+
+import scala.concurrent.Promise
+import scala.util.control.NonFatal
+
+class AsyncTcpClientObservable private[tcp] (
```

Comment:
we should try to isolate the similarities with AsyncFileReaderObservable in a trait and reuse them